pkg/proxy/endpoints.go

```diff
@@ -190,39 +190,54 @@ func (ect *EndpointChangeTracker) EndpointSliceUpdate(endpointSlice *discovery.E
 	ect.lock.Lock()
```

Comment:
nit: consider enclose the critical section into a built in func
```
func(){
  ect.lock.Lock()
  defer ect.lock.Unlock()

  ...
}()
```

Reply:
you have lock in endpointSliceCache already. Do you still need lock here?

Reply:
After discussing this a bit more I think it makes sense to hold off on any additional changes here until `endpointslicecache` can be moved into its own package.